src/modules/user/entry-points/http/user.controller.ts

```diff
@@ -16,4 +19,54 @@ export class UserController {
       user,
     };
   }
+
+  @Post("wallets")
+  @UseGuards(JwtAuthGuard)
+  @ApiTags("users")
+  public async postWallet(@Req() req: any, @Body() body: UsersWalletsBody) {
+    const userId = req.user.id;
+    const { walletAddress, signature, discordId } = body;
+
+    this.walletService.verifySignedDiscordId({
+      signature,
+      discordIdMessage: discordId,
+      walletAddress,
+    });
+
+    await this.walletService.assertUserExists(userId);
+
+    const wallet = await this.walletService.upsertWallet({
+      userId,
+      walletAddress,
+    });
+
+    return {
+      wallet,
+    };
```

Comment:
I feel like we need to keep the controllers structure as simple as possible. Ideally a particular HTTP handler would make a single call to a particular service. In our case it'd be something like `this.walletService.linkUserWallet(...)`

Reply:
done